packages/@orchidui-vue/src/Form/LinkInput/OcLinkInput.vue

```diff
@@ -30,63 +32,94 @@ const props = defineProps({
     type: Object,
     default: () => ({}),
   },
+  isEdit: Boolean,
 });
-defineEmits({
+
+const emit = defineEmits({
   "update:modelValue": [],
+  "update:type": [],
+  "update:title": [],
 });
+
 const isDropdownOpened = ref(false);
-const selectedLink = ref(props.links?.[0]?.value || "");
-const selectedLinkProps = computed(() =>
-  props.links.find((link) => link.value === selectedLink.value),
+const selectedLinkType = ref(props.type ?? props.links?.[0]?.value);
+const linkTitle = ref(props.title ?? "");
+const selectedLinkTypeProps = computed(() =>
+  props.links.find((link) => link.value === selectedLinkType.value),
 );
+const localValue = ref(
+  props.modelValue
+    ? props.modelValue.replace(selectedLinkTypeProps.value.preFill, "")
+    : "",
+);
+
+const updateLinkType = (value) => {
+  selectedLinkType.value = value;
+  emit("update:type", value);
+  isDropdownOpened.value = false;
+};
+
+const update = (value) => {
+  emit("update:modelValue", selectedLinkTypeProps.value.preFill + value);
+  if (!props.isEdit && selectedLinkType.value !== "link") {
+    emit("update:title", value);
+  }
+};
 </script>
 
 <template>
-  <Input
-    :placeholder="placeholder"
-    :label="label"
-    :error-message="errorMessage"
-    :is-inline-label="isInlineLabel"
-    :disabled="isDisabled"
-    :hint="hint"
-    :pre-fill="selectedLinkProps.preFill"
-    :model-value="modelValue"
-    :is-required="isRequired"
-    :label-icon="labelIcon"
-    :tooltip-text="tooltipText"
-    :tooltip-options="tooltipOptions"
-    @update:model-value="$emit('update:modelValue', $event)"
-  >
-    <template #trailing>
-      <Dropdown v-model="isDropdownOpened" :distance="10">
-        <div class="flex text-oc-text-400 items-center gap-x-2">
-          <Icon width="20" height="20" :name="selectedLinkProps.icon" />
-          <Icon
-            width="16"
-            height="16"
-            class="transition duration-300"
-            name="chevron-down"
-            :class="isDropdownOpened ? '-rotate-180' : ''"
-          />
-        </div>
+  <div>
+    <Input
+      v-if="selectedLinkType === 'link' || isEdit"
+      v-model="linkTitle"
+      class="mb-3"
+      label="Title"
+      placeholder="Title"
+      @update:model-value="$emit('update:title', $event)"
+    />
+    <Input
+      :placeholder="selectedLinkTypeProps.placeholder"
```

Comment:
here if `linkType` will be `'link' `and it will not be in `array of links` we will see error cause `selectedLinkTypeProps` in that case will be `undefined` should be some checking of `undefined` also this error will be in below in `preFill`

Reply:
Or should here be added v-else?

Reply:
![Screenshot 2023-12-26 at 16 14 28](https://github.com/hit-pay/orchid/assets/46804248/b4fea000-3cd8-4d3d-ac2c-5c2957c32fa5)

![Screenshot 2023-12-26 at 16 14 36](https://github.com/hit-pay/orchid/assets/46804248/394bf097-97e5-4a0d-967b-14ae5ce56c05)
 
no need v-else @mykhailo-diakovych , ui will be like this, and title using v-model:title

Reply:
custom link title only for type other website and edit mode.

Reply:
fixed